pkgs/development/python-modules/disnake/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, aiohttp
+, buildPythonPackage
+, fetchFromGitHub
+, libopus
+, pynacl
+, pythonOlder
+, withVoice ? true
+, ffmpeg
+}:
+
+buildPythonPackage rec {
+  pname = "disnake";
+  version = "2.9.2";
+  format = "setuptools";
```

Comment:
Could you set `pyproject = true` and add `setuptools` to `build-system` to avoid a legacy build?

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters